pkg/api/authz.go

```diff
@@ -89,14 +98,25 @@ func (ac *AccessController) can(username, action, repository string) bool {
 		}
 	}
 
+	acCtx, err := localCtx.GetAccessControlContext(ctx)
+	if err != nil {
+		return false
+	}
+
+	groups := acCtx.Groups
+
 	// check matched repo based policy
 	pg, ok := ac.Config.Repositories[longestMatchedPattern]
 	if ok {
-		can = isPermitted(username, action, pg)
+		can = ac.isPermitted(groups, username, action, pg)
 	}
 
 	// check admins based policy
 	if !can {
+		if ac.isAnyGroupInAdminPolicy(groups, username) && common.Contains(ac.Config.AdminPolicy.Actions, action) {
```

Comment:
First check username, and afterwards the group.

Reply:
I changed the order of the checks so that the username's permissions are checked first, then for the groups